addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4685,19 +4684,19 @@ export class OdooEditor extends EventTarget {
             const textSliced = selection.anchorNode.textContent.slice(0, selection.anchorOffset);
             const textNodeSplitted = textSliced.split(/\s/);
             const potentialUrl = textNodeSplitted.pop();
-            const match = potentialUrl.match(URL_REGEX);
+            // In case of multiple matches, only the last one will be converted.
+            const match = [...potentialUrl.matchAll(new RegExp(URL_REGEX, 'g'))].pop();
 
-            if (match && match[0] === potentialUrl && !EMAIL_REGEX.test(potentialUrl)) {
+            if (match && !EMAIL_REGEX.test(match[0])) {
+                const nodeForSelectionRestore = selection.anchorNode.splitText(selection.anchorOffset);
                 const url = match[2] ? match[0] : 'http://' + match[0];
```

Comment:
@sebgeelen Here on this unchanged line you can see that the default protocol is `http`.